packages/headless/src/composables/menuButton/createMenuButton.ts

```diff
@@ -110,8 +111,10 @@ export const createMenuButton = createBuilder((options: CreateMenuButtonOptions)
       root: {
         id: rootId,
         onKeydown: handleKeydown,
-        onMouseenter: () => setExpanded(true),
-        onMouseleave: () => setExpanded(false, true),
+        onMouseenter: () => (options.trigger?.value === "hover" ? setExpanded(true) : undefined),
+        onMouseleave: () =>
+          options.trigger?.value === "hover" ? setExpanded(false, true) : undefined,
+        onClick: () => (options.trigger?.value === "click" ? setExpanded(true) : undefined),
```

Comment:
```suggestion
        onMouseenter: options.trigger?.value === "hover" ? () => setExpanded(true) : undefined,
        onMouseleave: options.trigger?.value === "hover" ? () => setExpanded(false, true) : undefined,
        onClick: options.trigger?.value === "click" ? () => setExpanded(true) : undefined,
```

Would prefer to not add the event listeners at all instead of having empty event listeners